src/components/users-list.tsx

```diff
@@ -1,18 +1,37 @@
 import {useQuery} from '@apollo/client';
-import React from 'react';
-import {
-  ActivityIndicator,
-  FlatList,
-  ListRenderItemInfo,
-  SafeAreaView,
-  Text,
-} from 'react-native';
+import React, {useState} from 'react';
+import {ActivityIndicator, FlatList, SafeAreaView, Text} from 'react-native';
 import {usersQuery} from '../utils/apollo';
 import {UserNode} from '../utils/models';
-import {UserItem} from './user-item';
+import {Loading, renderItem} from './list-functions';
+
+const limit = 15;
 
 export const UsersList = () => {
-  const {data, loading} = useQuery(usersQuery);
+  const [offset, setOffset] = useState(0);
+  const [hasMoreData, setHasMoreData] = useState(true);
+  const [list, setList] = useState<UserNode[]>([]);
```

Comment:
```suggestion
  const [users, setUsers] = useState<UserNode[]>([]);
```

Mais claro assim, né?